src/stash/tests/stash.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashSet;
+use std::error::Error;
+
+use ore::metrics::MetricsRegistry;
+use tempfile::{NamedTempFile, TempDir};
+
+use mz_stash::{PersistStash, SqliteStash, Stash, StashOp};
+use persist::file::{FileBlob, FileLog};
+use persist::storage::{Blob, LockInfo};
+
+#[test]
+fn test_sqlite() -> Result<(), anyhow::Error> {
+    let file = NamedTempFile::new()?;
+    run_tests(|| SqliteStash::open(file.path()))?;
+    Ok(())
+}
+
+#[test]
+fn test_persist() -> Result<(), anyhow::Error> {
+    let dir = TempDir::new()?;
+    run_tests(|| {
+        let lock_info = LockInfo::new("stash".into(), "nonce".into())?;
+        let log = FileLog::new(dir.path().join("log"), lock_info.clone())?;
+        let blob = FileBlob::open_exclusive(dir.path().join("blob").into(), lock_info)?;
```

Comment:
unless you're looking for concrete performance numbers, MemBlob is designed exactly for tests. MemRegistry allows you to share a MemBlob across restarts of persist. concretely, i care because MemBlob will make this unit test much faster under `cargo stress`